code/game/objects/items/weapons/holy_weapons.dm

```diff
@@ -113,14 +113,143 @@
 
 /obj/item/nullrod/godhand
 	name = "god hand"
-	icon_state = "disintegrate"
-	item_state = "disintegrate"
-	desc = "This hand of yours glows with an awesome power!"
+	icon_state = "burnhand"
+	item_state = "burnhand"
+	desc = "Like Prometheus, you steal fire from the Gods! Except you didn't steal this."
 	flags = ABSTRACT | NODROP| DROPDEL
 	w_class = WEIGHT_CLASS_HUGE
 	hitsound = 'sound/weapons/sear.ogg'
+	force = 15
 	damtype = BURN
-	attack_verb = list("punched", "cross countered", "pummeled")
+	light_color = "#E09D37"
+	attack_verb = list("punched", "pummeled", "burnt", "seared")
+			
+/obj/item/nullrod/godhand/attack(mob/living/carbon/M, mob/user)
+	if(!isliving(M))
+		return
+	M.IgniteMob()
+	if(!ismob(M))
+		return
+
+	if(istype(M.wear_mask, /obj/item/clothing/mask/cigarette) && user.zone_selected == "mouth")
+		var/obj/item/clothing/mask/cigarette/cig = M.wear_mask
+		if(M == user)
+			cig.attackby(src, user)
+		else
+			if(istype(src, /obj/item/nullrod/godhand))
+				cig.light("<span class='rose'>[user] extends [user.p_their()] hand out to [M], and [cig] ignites.</span>")
+			else
+				cig.light("<span class='notice'>[user] reaches out to [M] with [user.p_their()] hand, and [cig] suddenly ignites.</span>")
```

Comment:
This first part will always be true. You will never reach the `cig.light("<span class='notice'>[user] reaches out to [M] with [user.p_their()] hand, and [cig] suddenly ignites.</span>")` statement.